+precice/@SolverInterface/SolverInterface.m

```diff
@@ -150,18 +150,20 @@ function markActionFulfilled(obj,action)
         
         % setMeshVertices
         function vertexIds = setMeshVertices(obj,meshID,positions)
+            assert(size(positions, 1) ~=  obj.getDimensions(), 'The shape of the matrices must be [dim numVertices], where dim is the problem dimension');
```

Comment:
Why `~=` ? Shouldn't it be `==`?

Reply:
The error `'The shape of the matrices must be [dim numVertices], where dim is the problem dimension'` is always the same. Factor this out into a function to avoid code duplication. Example: `checkDimensions(size(positions, 1), obj.getDimensions())`, where the first is the actual dimension and the second the expected dimensions.

Reply:
You're totally right, sorry. I was using an `if ~=` for throwing an error, and when changing to assertions I didn't realized I did not change it.

Reply:
In this case the LC tutorial being a 0-dimensional problem does not need the multiple-vertices preCICE functions, and as such the error was not catched by the testing procedure. 
This suggests that the current testing procedures using the solverdummy and the LC tutorial are maybe not enough as they do not test the whole API, and unit testing is necessary.